microprot/snakemake/Snakefile

```diff
@@ -3,192 +3,276 @@
+                                                                frag_len,
+                                                                p_val,
+                                                                e_val,
+                                                                prob))
+        split_search.mask_sequence(hh_inp, fasta_inp,
+                                   subsequences_fp='%s/%s' % (outpath,
+                                                              outname),
+                                   min_fragment_length=frag_len,
+                                   max_evalue=e_val,
+                                   min_prob=prob,
+                                   max_pvalue=p_val
+                                   )
+        match = '%s/%s.match' % (outpath, outname)
+        non_match = '%s/%s.non_match' % (outpath, outname)
 
-SEQUENCES=config['SEQUENCES']
-# SEQUENCES = glob_wildcards(paths['INPUTS'])
+        if snakemake_helpers.not_empty(match):
+            snakemake_helpers.append_db(match, step=step, version=version,
+                                        db_fp=str(paths['MICROPROT_DB']))
+        # TODO
+        # if non_match is not empty `touch $out.ok`, else `touch $out.pass`?
+        # will this help avoid fails? i.e. in subsequent rules, if `ok` is
+        # present, execte rule, if `pass`, just skip execution and pass on
+        # another dummy file
+        if snakemake_helpers.not_empty(non_match):
+            shell('touch {out_0}')
 
-def search_X(seq, split, output_base, shell_func):
-    for d_boundaries, domain in split_sequence(seq, split):
-        n = '%s-%s' % (str(d_boundaries[0]), str(d_boundaries[1]))
-        out = '%s_%s.%s' % (output_base, n, 'out')
-        a3m = '%s_%s.%s' % (output_base, n, 'a3m')
-        with open(output_base, 'a') as f:
-            f.write(out)
-            f.write(a3m)
-        shell_func
-        pass
 
 rule all:
     input:
-        expand("results/{seq}/10-Rosetta/{seq}.msa",
-               seq=SEQUENCES),
-        expand("results/{seq}/03-secondary_predictions/{seq}_{n}.{suffix}",
-               seq=SEQUENCES,
-               n=['1-100', '101-250', '251-400'],
-               suffix=['ss', 'diso', 'tmh'])
-    run:
-        for inp in input:
-            inp = re.sub('{seq}/10-Rosetta/', '', inp)
-            inp = re.sub('results/', 'results/log/', inp)
-            inp = re.sub('.msa', '.params', inp)
-            with open(inp, 'w') as yaml_file:
-                yaml.dump(config, yaml_file, default_flow_style=False)
-                yaml.dump(paths, yaml_file, default_flow_style=False)
-
+        expand(config['MICROPROT_MAIN']+'results/log/{seq}/msa_ripe.log',
+               seq=SEQ_ids)
 
-rule extract_sequence:
+rule read_sequences:
     input:
-        SEQUENCES
+        '%s/%s' % (SEQ_path, SEQ_file)
     output:
-        'data/{seq}.fasta'
+        config['MICROPROT_MAIN']+'data/WCG_inputs/{seq}.fasta'
```

Comment:
output filepaths depend on the "name" of the sequence. What if this name contains strange characters as / or | ?